### PR TITLE
Add bare host boundary tests

### LINKED ISSUE #449: Add boundary tests for host rewrite bare-host substitution
## Description

`rewrite_bare_host_at_boundaries` in `crates/common/src/host_rewrite.rs` has complex boundary detection logic and no direct tests. This leaves host rewriting vulnerable to regressions around subdomains, punctuation, word boundaries, and host:port forms.

## Done when

- [ ] Add unit tests for exact host matches at valid boundaries.
- [ ] Add negative tests for subdomains and embedded-word partial matches.
- [ ] Add coverage for multiple occurrences, empty input, and host:port cases.
- [ ] Keep the test names focused on rewrite semantics rather than implementation details.

## Affected area

- HTML processing / JS injection

## Additional context

Source: `test_gap.md`, section `URL Rewriting Boundary Detection`.
Parent issue: #396

### PR DESCRIPTION
**Pull request by ChristianPavilonis**
## Summary

- Adds direct unit coverage for bare-host rewrite boundary behavior.
- Fixes standalone `host:port` bare-host rewrites while preserving the port and following path/query/fragment.
- Locks down negative cases for subdomains, embedded tokens, larger hostnames, and invalid colon suffixes.

## Changes

| File | Change |
| ---- | ------ |
| `crates/trusted-server-core/src/host_rewrite.rs` | Documented numeric port handling, allowed valid `:port` suffixes as safe rewrite boundaries, and added focused unit tests for boundary behavior. |

## Closes

Closes #449

## Test plan

- [x] `cargo test --workspace`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo fmt --all -- --check`
- [ ] JS tests: `cd crates/js/lib && npx vitest run`
- [ ] JS format: `cd crates/js/lib && npm run format`
- [ ] Docs format: `cd docs && npm run format`
- [ ] WASM build: `cargo build --package trusted-server-adapter-fastly --release --target wasm32-wasip1`
- [ ] Manual testing via `fastly compute serve`
- [x] Other: `cargo test -p trusted-server-core host_rewrite`; `cargo test -p trusted-server-core nextjs::shared`; `cargo test -p trusted-server-core rsc_flight`; `cargo test -p trusted-server-core`

## Checklist

- [x] Changes follow [CLAUDE.md](/CLAUDE.md) conventions
- [x] No `unwrap()` in production code — use `expect("should ...")`
- [x] Uses `tracing` macros (not `println!`)
- [x] New code has tests
- [x] No secrets or credentials committed

### CODE REVIEW

File: crates/trusted-server-core/src/host_rewrite.rs [hunk 1]
```diff
@@ -47,3 +64,99 @@ pub(crate) fn rewrite_bare_host_at_boundaries(
...
+            rewrite(input),
+            None,
+            "should not rewrite subdomains, embedded words, or larger hostname tokens"
+        );
+    }
+
+    #[test]
+    fn rewrites_host_with_valid_numeric_port() {
+        let input = "origin.example.com:8443/path origin.example.com:9443?x=1 origin.example.com:443#frag origin.example.com:8080";
+        let expected = "proxy.example.com:8443/path proxy.example.com:9443?x=1 proxy.example.com:443#frag proxy.example.com:8080";
```
Comment (aram356) [inline crates/trusted-server-core/src/host_rewrite.rs]: ♻️ **refactor** — The positive port test only exercises `/`, `?`, `#`, and end-of-string after the port, but `has_valid_port_boundary` accepts *any* non-host separator. Pin the common wrapped `host:port` case (paren-wrapped here) so the broad boundary acceptance is intentional and regression-guarded.

```suggestion
        let input = "origin.example.com:8443/path origin.example.com:9443?x=1 origin.example.com:443#frag origin.example.com:8080 (origin.example.com:5000)";
        let expected = "proxy.example.com:8443/path proxy.example.com:9443?x=1 proxy.example.com:443#frag proxy.example.com:8080 (proxy.example.com:5000)";
```

File: crates/trusted-server-core/src/host_rewrite.rs [hunk 2]
```diff
@@ -47,3 +64,99 @@ pub(crate) fn rewrite_bare_host_at_boundaries(
...
+
+    #[test]
+    fn does_not_rewrite_larger_hostname_tokens() {
+        let input = "cdn.origin.example.com notorigin.example.com origin.example.com.evil origin.example.com-extra origin.example.comextra";
+
+        assert_eq!(
+            rewrite(input),
+            None,
+            "should not rewrite subdomains, embedded words, or larger hostname tokens"
+        );
```
Comment (aram356) [inline crates/trusted-server-core/src/host_rewrite.rs]: ⛏ **nitpick** — These five negative cases share a single `None` assertion, so a regression on one token won't tell you which. Splitting into per-case assertions keeps the failure localized. Non-blocking.

```suggestion
        assert_eq!(
            rewrite("cdn.origin.example.com"),
            None,
            "should not rewrite a subdomain prefix"
        );
        assert_eq!(
            rewrite("notorigin.example.com"),
            None,
            "should not rewrite an embedded-word prefix"
        );
        assert_eq!(
            rewrite("origin.example.com.evil"),
            None,
            "should not rewrite a trailing-dot label"
        );
        assert_eq!(
            rewrite("origin.example.com-extra"),
            None,
            "should not rewrite a trailing-hyphen token"
        );
        assert_eq!(
            rewrite("origin.example.comextra"),
            None,
            "should not rewrite a trailing-word token"
        );
```

File: crates/trusted-server-core/src/host_rewrite.rs [hunk 3]
```diff
@@ -47,3 +64,99 @@ pub(crate) fn rewrite_bare_host_at_boundaries(
...
+
+    #[test]
+    fn does_not_rewrite_invalid_port_like_suffixes() {
+        let input = "origin.example.com:not-a-port origin.example.com:8443evil origin.example.com:8443.evil origin.example.com:";
+
+        assert_eq!(
+            rewrite(input),
+            None,
+            "should not treat arbitrary colon suffixes as host boundaries"
+        );
```
Comment (aram356) [inline crates/trusted-server-core/src/host_rewrite.rs]: ⛏ **nitpick** — Same as the larger-hostname test: split the bundled colon-suffix cases so a single regression is identifiable. Non-blocking.

```suggestion
        assert_eq!(
            rewrite("origin.example.com:not-a-port"),
            None,
            "should not treat a non-numeric suffix as a port"
        );
        assert_eq!(
            rewrite("origin.example.com:8443evil"),
            None,
            "should not treat a port with a trailing word as a boundary"
        );
        assert_eq!(
            rewrite("origin.example.com:8443.evil"),
            None,
            "should not treat a port with a trailing dot as a boundary"
        );
        assert_eq!(
            rewrite("origin.example.com:"),
            None,
            "should not treat an empty port as a boundary"
        );
```

**Review by aram356 — APPROVED** [review]
## Summary

Adds focused unit coverage for `rewrite_bare_host_at_boundaries` (closing #449) plus a small production fix: a numeric `:port` suffix is now treated as a valid host boundary, so a standalone `host:port` is rewritten with the port preserved. The change is correct (the `end == bytes.len()` short-circuit guards the `bytes[end]` index; slicing stays on ASCII boundaries) and makes the bare-host path consistent with the URL path, which already preserves ports via `strip_origin_host_with_optional_port`.

Approving. The three inline comments are non-blocking polish carried as one-click `suggestion` blocks — all scratch-verified against the PR head (`cargo fmt --check`, `cargo test -p trusted-server-core host_rewrite`, `cargo clippy -p trusted-server-core --all-targets --all-features`).

> 3 of the inline comments below carry a one-click GitHub `suggestion`.

## Non-blocking

### ♻️ refactor / ⛏ nitpick

- **Cover the generic-separator port boundary** — see inline at `crates/trusted-server-core/src/host_rewrite.rs:142-143`
- **Split bundled larger-hostname negatives** — see inline at `crates/trusted-server-core/src/host_rewrite.rs:131-137`
- **Split bundled invalid-port negatives** — see inline at `crates/trusted-server-core/src/host_rewrite.rs:154-160`

## CI Status

- cargo test: PASS (required)
- cargo fmt: PASS (required)
- format-typescript: PASS (required)
- format-docs: PASS (required)
- Analyze (rust): PASS
- Analyze (javascript-typescript): PASS
- Analyze (actions): PASS
- CodeQL: PASS
- vitest: PASS
- integration tests: PASS
- browser integration tests: PASS
- prepare integration artifacts: PASS

**Review by prk-Jr — APPROVED** [review]
